library/spdm_requester_lib/libspdm_req_send_receive.c

```diff
@@ -601,6 +601,18 @@ libspdm_return_t libspdm_send_spdm_request(libspdm_context_t *spdm_context,
         }
     }
 
+    if ((spdm_context->connection_info.capability.max_spdm_msg_size != 0) &&
+        (request_size > spdm_context->connection_info.capability.max_spdm_msg_size)) {
+        LIBSPDM_DEBUG((LIBSPDM_DEBUG_ERROR, "request_size > rsp max_spdm_msg_size\n"));
+        return LIBSPDM_STATUS_PEER_BUFFER_TOO_SMALL;
+    }
+    if (request_size > spdm_context->local_context.capability.max_spdm_msg_size) {
```

Comment:
`local_context.capability.max_spdm_msg_size` is the same size / buffer for both sending and receiving message, correct?

Reply:
correct.
We only distinguish device hw buffer for sender/receiver. 
max_spdm_msg_size is sw buffer and same for sender/receiver.

Reply:
Ok. I would replace all of this with
```
LIBSPDM_ASSERT(request_size <= spdm_context->local_context.capability.max_spdm_msg_size);
```
since the request message has already been formed and by definition it can't be larger than the local `max_spdm_msg_size`.

Reply:
done.